azext_hanaonazure/modules_sdk/models/hana_instance_py3.py

```diff
@@ -78,10 +86,12 @@ class HanaInstance(Resource):
         'power_state': {'key': 'properties.powerState', 'type': 'str'},
         'proximity_placement_group': {'key': 'properties.proximityPlacementGroup', 'type': 'str'},
         'hw_revision': {'key': 'properties.hwRevision', 'type': 'str'},
+        'partner_node_id': {'key': 'properties.partnerNodeId', 'type': 'str'},
+        'provisioning_state': {'key': 'properties.provisioningState', 'type': 'str'},
     }
 
-    def __init__(self, *, hardware_profile=None, storage_profile=None, os_profile=None, network_profile=None, **kwargs) -> None:
-        super(HanaInstance, self).__init__(**kwargs)
+    def __init__(self, *, location: str=None, hardware_profile=None, storage_profile=None, os_profile=None, network_profile=None, partner_node_id: str=None, **kwargs) -> None:
```

Comment:
I see all of these parameters have default value of None. So all of these are optional?

Reply:
Yes; it's up to us as an RP to reject things missing required data. From the CLI, we can specify required parameters and then fill out these structures like `hardware_profile`.

Note: everything under azext_hanaonazure/modules_sdk/* is auto-generated, not written by me